### ISSUE TITLE
ERROR NOK: ("can't start new thread",) with certail jails

### ISSUE THREAD

**Issue by AlphaJack** [fail2ban/fail2ban#2922]
### Environment:

- Fail2Ban version (including any possible distribution suffixes): 0.11.2
- OS, including release name/version:  5.10.11-3-ARCH #1 SMP Fri Jan 29 18:07:27 UTC 2021 armv7l GNU/Linux
- [x] Fail2Ban installed via OS/distribution mechanisms
- [x] You have not applied any additional foreign patches to the codebase
- [x] Some customizations were done to the configuration (provide details below is so)

### The issue:
If I add one of the three offending jails and I reload the client, fail2ban stop working showing a long line: "WARNING Command [...] Received RuntimeError("can't start new thread")". The omitted part is a long list of actions fail2ban does. 

#### Steps to reproduce
1. start with a clean jail.local, no problem
2. add some jails and reload, no problem
3. add one of these jails, you get
```
Job for fail2ban.service failed.
See "systemctl status fail2ban.service" and "journalctl -xe" for details.
```
4. remove the jail, no problem

#### Expected behavior
All jails should work as they always did

#### Observed behavior
See below

#### Any additional information

The strange thing is that when I enabled these jails the first time I did not face this problem. In the meanwhile I have updated my kernel and rebooted many times, but only since this week the problem appeared.

### Configuration, dump and another helpful excerpts

#### Any customizations done to /etc/fail2ban/ configuration
`postfix.conf` filter:
```
No change
```
`radicale.conf` filter
```ini
[Definition]                                                                                                                                                                                                                                                                          
failregex = ^.* Failed login attempt from '<ADDR>'
[Init]
journalmatch = _SYSTEMD_UNIT=radicale.service                                                                                                                                                                         
```

`jail.local`
```ini
[DEFAULT]
ignoreip = 127.0.0.1/8 192.168.1.0/24
banaction = nftables-allports
chain = input
bantime = 90d
findtime = 1
maxretry = 1
loglevel = 4

# working jails
[...]

# failing jails

[postfix]
enabled = true
mode = aggressive

[postfix-slow]
enabled = true
backend = systemd
filter = postfix[mode = aggressive]
maxretry = 5
findtime = 1800

[radicale]
enabled = true
backend = systemd
```

#### Relevant parts of /var/log/fail2ban.log file:

`systemd unit`:
```
Jan 30 20:41:29 hostname systemd[1]: Reloading Fail2Ban Service.
Jan 30 20:41:35 hostname fail2ban-client[5321]: 2021-01-30 20:41:35,804 fail2ban                [5321]: ERROR   NOK: ("can't start new thread",)
Jan 30 20:41:35 hostname fail2ban-client[5321]: can't start new thread
Jan 30 20:41:36 hostname systemd[1]: fail2ban.service: Control process exited, code=exited, status=255/EXCEPTION
```

`fail2ban log`:
```
2021-01-30 20:48:04,576 fail2ban.server         [6378]: INFO    Jail 'XXXXXX' reloaded                                                                                                                                                                          
2021-01-30 20:48:04,609 fail2ban.server         [6378]: INFO    Reload finished.                                                                                                                                                                                      
2021-01-30 20:48:04,610 fail2ban.transmitter    [6378]: WARNING Command ['reload', '--all', [], [['set', 'syslogsocket', 'auto'], ['set', 'loglevel', 'INFO'], ['set', 'logtarget', '/var/log/fail2ban.log'], ['set', 'dbfile', '/var/lib/fail2ban/fail2ban.sqlite3'],
 ['set', 'dbmaxmatches', 10], ['set', 'dbpurgeage', '1d'], ['add', 'XXXXXX', [...] ['nftables_family', 'inet'], ['nftables_table', 'fail2ban'], ['nftables_set_prefix', '']]], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX'], ['start', 'XXXXXX']]] has failed. Received RuntimeError("can't start new thread")
 ```

**Comment by AlphaJack** [comment]
IDK how, but I think I've fixed it
`postfix.local`
```ini
mode = aggressive
```

`jail.local`
```ini
[...]

[postfix]
enabled = true

[postfix-slow]
enabled = true
filter = postfix
maxretry = 5
findtime = 1800

[radicale]
enabled = true
backend = systemd

[...]
```
Again, everthing worked when I set up these jails like in my first post, no idea of what happened

**Comment by sebres** [comment]
You have to set an appropriate `stacksize` in fail2ban.local (seems to be too high or low on your system by default), see #2356 and because systemd is involved too as well as #969 for similar issue.

**Comment by AlphaJack** [comment]
The problem doesn't occur anymore with Radicale jail, but only with Postfix ones: if I add one of them to a working configuration, suddenly the reload operation takes much more time and eventually fails. If I remove the jail and try to reload the conf, systemd says "failed" as fail2ban log returns:
```
ERROR   Stop <Actions(f2b/a.postfix, initial daemon)> of jail 'postfix' failed: cannot join thread before it is started
``` 

If I stop the service and restart it with no Postfix jails it seems to run properly, but in the log file I always get the long warning between jail initialization and ban restoring:
```
[...]
INFO    Creating new jail 'nginx-auth'
INFO    Jail'nginx-4xx' started

"WARNING Command 
    [...]
Received RuntimeError("can't start new thread")"
ERROR   NOK: ("can't start new thread",)

NOTICE  [nginx-auth] Restore Ban 178.20.226.226
[...]
```

If I stop the service and restart it with Postfix jails, systemd doesn't report any error, but in the long file I always get the long warning at the end of the file, and I don't see any ban restored (no jail is being started):

```
[...]
INFO    Creating new jail 'postfix'

"WARNING Command 
    [...]
Received RuntimeError("can't start new thread")"
ERROR   NOK: ("can't start new thread",)
```

Same problem using the default Postfix filter with default mode

**Comment by sebres** [comment]
Also note you have to **restart** fail2ban if you're changing thread stacksize, a **reload would not work**, because this setting must affect before all threads going started (at begin of the start process) and reload would not restart the jail threads.

**Comment by AlphaJack** [comment]
Do I need to rebuild fail2ban from the latest commit or is it enough to specify
```ini
[Thread]
stacksize = 32
```
in jail.local?

**Comment by sebres** [comment]
> Do I need to rebuild fail2ban from the latest commit 

as you can see 3c70fe298a55f587b8a5633ad775ebfc19b83c95 is released in 0.11.2, 0.11.1, 0.10.6 and 0.10.5

>  is it enough to specify ... in jail.local?

No, it belongs to `fail2ban.local`.

https://github.com/fail2ban/fail2ban/blob/3c70fe298a55f587b8a5633ad775ebfc19b83c95/config/fail2ban.conf#L71-L77
